### PR TITLE
Beams will now properly center on things with pixel_x and pixel_y offsets

### PR DESCRIPTION
**Pull request by mikomyazaki**
- Charging cables won't be able to appear for 1 tick when a connection was attempted beyond their max range anymore.

- Cleared up a bunch of unnecessary checks/maths, and made it properly center on things that aren't in the center of turfs (e.g. APCs with their offset which is important for charging cables)

- Also changed some var names because `X.pixel_x = Pixel_x` made me sad.

**Current Behavior**
pda_charging_cable_offset

**New Behavior**
reely_good

### CODE REVIEW

**Comment by mikomyazaki** [comment]
!bugfix

**Comment by mikomyazaki** [comment]
!review

**Review by NonQueueingMatt — APPROVED** [review]

File: code/datums/beam.dm [hunk 1]
```diff
@@ -88,7 +94,7 @@
 	return ..()
 
 /datum/beam/proc/Draw()
-	var/Angle = round(Get_Angle(get_turf(origin), get_turf(target)))
+	var/Angle = round(Get_Angle(origin.x ? origin : get_turf(origin), target.x ? target : get_turf(target)))
```
Comment (Amunak) [inline code/datums/beam.dm:97]: Could `origin` or `target` possibly still be `null` here (perhaps when it's not set in `new`?

I'd consider this just to be sure:
```suggestion
	var/Angle = round(Get_Angle(origin?.x ? origin : get_turf(origin), target?.x ? target : get_turf(target)))
```

**Review by Amunak — APPROVED** [review]

File: code/datums/beam.dm (hunk 1 above)
Comment (mikomyazaki) [reply code/datums/beam.dm:97]: If it *is* `null` then something has been called improperly and we want the runtime to happen. `target` and `origin` are set in `/datum/beam/New()`

Also if we do `target?.x ? target : get_turf(target)` then the ternary might try to call `get_turf(target)` when `target` is null... which is going to have the same result (runtime).

**Review by Geevies — APPROVED** [review]

File: code/datums/beam.dm (hunk 1 above)
Comment (Amunak) [reply code/datums/beam.dm:97]: Fair enough. 👍

**Pull request merged**